soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -999,14 +1006,21 @@ void Randomizer::ParseItemLocationsFile(const char* spoilerFileName, bool silent
                     // todo handle prices
                     if (itemit.key() == "item") {
                         gSaveContext.itemLocations[randomizerCheck].check = randomizerCheck;
-                        gSaveContext.itemLocations[randomizerCheck].get = SpoilerfileGetNameToEnum[itemit.value()];
+                        gSaveContext.itemLocations[randomizerCheck].get.rgID = SpoilerfileGetNameToEnum[itemit.value()];
                     } else if (itemit.key() == "price") {
                         merchantPrices[gSaveContext.itemLocations[randomizerCheck].check] = itemit.value();
+                    } else if (itemit.key() == "model") {
+                        gSaveContext.itemLocations[randomizerCheck].get.fakeRgID =
+                            SpoilerfileGetNameToEnum[itemit.value()];
+                    } else if (itemit.key() == "trickName") {
+                        memcpy(gSaveContext.itemLocations[randomizerCheck].get.trickName,
+                               std::string(itemit.value()).c_str(), std::string(itemit.value()).length());
```

Comment:
we might want to use `strncpy` here instead of `memcpy`, and maybe just always copy `50` instead of the itemit length to ensure we don't end up writing places we don't want to in the odd case we get a wildly wrong value for itemit

Reply:
Ok that makes sense, will do when I'm back home tonight.